src/zero_copy/text.rs

```diff
@@ -59,32 +95,297 @@ impl Char for u8 {
             .map(|m| m.end())
     }
 
+    fn from_ascii(c: u8) -> Self {
+        c
+    }
+    fn is_inline_whitespace(&self) -> bool {
+        *self == b' ' || *self == b'\t'
+    }
     fn is_whitespace(&self) -> bool {
         self.is_ascii_whitespace()
     }
+    fn digit_zero() -> Self {
+        b'0'
+    }
+    fn is_digit(&self, radix: u32) -> bool {
+        (*self as char).is_digit(radix)
+    }
+    fn to_char(&self) -> char {
+        *self as char
+    }
 }
 
 #[derive(Clone)]
 pub struct Padded<A> {
     pub(crate) parser: A,
 }
 
-impl<'a, I, E, S, A> Parser<'a, I, E, S> for Padded<A>
+impl<'a, I, O, E, S, A> Parser<'a, I, O, E, S> for Padded<A>
 where
     I: Input + ?Sized,
     E: Error<I>,
     S: 'a,
     I::Token: Char,
-    A: Parser<'a, I, E, S>,
+    A: Parser<'a, I, O, E, S>,
 {
-    type Output = A::Output;
-
-    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E, S>) -> PResult<M, Self::Output, E> {
+    fn go<M: Mode>(&self, inp: &mut InputRef<'a, '_, I, E, S>) -> PResult<M, O, E> {
         inp.skip_while(|c| c.is_whitespace());
         let out = self.parser.go::<M>(inp)?;
         inp.skip_while(|c| c.is_whitespace());
         Ok(out)
     }
 
-    go_extra!();
+    go_extra!(O);
+}
+
+/// A parser that accepts (and ignores) any number of whitespace characters.
+///
+/// This parser is a `Parser::Repeated` and so methods such as `at_least()` can be called on it.
+///
+/// The output type of this parser is `Vec<()>`.
+///
+/// # Examples
+///
+/// ```
+/// # use chumsky::zero_copy::prelude::*;
+/// let whitespace = text::whitespace::<_, _, Simple<str>>();
+///
+/// // Any amount of whitespace is parsed...
+/// assert_eq!(whitespace.parse("\t \n  \r k").0, Some("\t \n  \r "));
+/// // ...including none at all!
+/// assert_eq!(whitespace.parse("").0, Some(""));
+/// ```
+pub fn whitespace<'a, C: Char, I: StrInput<C> + ?Sized, E: Error<I>>(
+) -> impl Parser<'a, I, &'a I::Slice, E>
+where
+    I::Token: Char,
+{
+    any()
+        .filter(|c: &I::Token| c.is_whitespace())
+        .repeated()
+        .map_slice(|s| s)
+}
```

Comment:
Given the existence of `pythonic.rs` and other stuff that implies support for semantic indentation, perhaps an `inline_whitespace` parser to complement this would be nice? Not required for this PR though.